src/DFG.cpp

```diff
@@ -49,6 +38,75 @@ DFG::DFG(Function& t_F, list<Loop*>* t_loops, bool t_targetFunction,
 
 }
 
+// Specilized fusion for the nonlinear operations.
+void DFG::nonlinear_combine() {
+  tuneForBitcast();
+  combineMulAdd(2);
+  combinePhiAdd(1);
```

Comment:
Why we cannot use `combine("phi", "add", 1);`? because the `combine()` algorithm is too naive to resolve/combine what we want?

Reply:
Here I combine phi+add+add and phi+add and there is only integer addition considered, thus I created a new function to handle this special patterns.